lib/Graph/Graph.cpp

```diff
@@ -62,10 +62,22 @@ void Module::strip() {
 }
 
 void Module::clear() {
-  eraseFunctions();
-
   for (auto it = constants_.begin(), e = constants_.end(); it != e; it++) {
     Constant *v = *it;
+
+    // Log constant deletion in all its user functions.
+    FunctionList userFuncs;
+    for (auto use : v->getUsers()) {
+      auto userFunc = use.getUser()->getParent();
```

Comment:
Nit: Consider using `auto *` here to make it clearer that `userFunc` is a pointer and that you're not maintaining and searching through a list of entire objects.